addons/stock/models/stock_quant.py

```diff
@@ -423,6 +423,8 @@ def _update_available_quantity(self, product_id, location_id, quantity, lot_id=N
                             quant._create_or_update_neg_inv('add')
                         elif previously_neg:
                             quant._create_or_update_neg_inv('remove')
+                        if quant.product_id.tracking == 'serial' and lot_id:
```

Comment:
Same as conflicting inventories on negative quantities, only keep the scheduler part